site/app/views/submission/HomeworkView.php

```diff
@@ -371,23 +371,25 @@ private function renderSubmitBox(Gradeable $gradeable, $graded_gradeable, $versi
             }
         }
         else {
-            // Get path to VCS_CHECKOUT
-            $gradeable_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "submissions", $gradeable->getId());
-            $who_id = $this->core->getUser()->getId();
-            $user_path = FileUtils::joinPaths($gradeable_path, $who_id);
-            $highest_version = $graded_gradeable->getAutoGradedGradeable()->getHighestVersion();
-            $version_path = FileUtils::joinPaths($user_path, $highest_version);
-            $path = FileUtils::joinPaths($version_path, ".submit.VCS_CHECKOUT");
-
-            // Load repo and user id
-            if (file_exists($path)) {
-                $json = json_decode(file_get_contents($path), true);
-                if (!is_null($json)) {
-                    if (isset($json["git_user_id"])) {
-                        $github_user_id = $json["git_user_id"];
-                    }
-                    if (isset($json["git_repo_id"])) {
-                        $github_repo_id = $json["git_repo_id"];
+            if ($graded_gradeable !== null) {
+                // Get path to VCS_CHECKOUT
+                $gradeable_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "submissions", $gradeable->getId());
```

Comment:
Github won't let me comment on non-PR lines, so I'll put this comment here. 
* Line 322 looks as though it may still crash if `graded_gradeable` is null.
* Same for 478.

Reply:
I've added null checks on graded_gradeable for these two lines